### PR TITLE
[azure][trace] Refine error message when setup Cosmos DB

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR refines the process setuping Cosmos DB. Use Azure standard async request way to setup Cosmos DB, so that we can know concrete status of the setup status. Below are `pf config set trace.destination` for 3 scenarios (failed, already setup, setup):

image

Note that in this PR, as DDL is near, I implement the poll process with copy & modify; I have opened a task tracking the work that uses `LROPoller`, the standard way for async API request.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py [hunk 1]
```diff
@@ -757,3 +762,98 @@ def init_workspace_cosmos(
...
+            build_setup_trace_session_async_request,
+        )
+
+        headers = self._get_headers()
+        request_id = headers["x-ms-client-request-id"]
+        # below lines are copied and modified from TraceSessionsOperations.setup_trace_session_async
+        error_map = {401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError}
+        content_type = kwargs.pop("content_type", "application/json")
+        _json = self.caller.trace_sessions._serialize.body(body, "TraceDbSetupRequest")
+        request = build_setup_trace_session_async_request(
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py:789]: why do we need to put such code in service caller?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py:789]: this API is an async API and we need to poll; the most standard way to do this is using `LROPoller` in `azure-core`, but it needs more time to implement. so copy some codes from generated rest client here to get async poll url from response header.

File: src/promptflow-azure/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # promptflow-azure package
 
+## v1.11.0 (Upcoming)
```
Comment (brynn-code) [inline src/promptflow-azure/CHANGELOG.md:3]: add the promptflow changelog also

Comment (zhengfeiwang) [reply src/promptflow-azure/CHANGELOG.md:3]: updated

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**